spec/unit/knife/winrm_spec.rb

```diff
@@ -443,6 +443,22 @@
       expect { @winrm.run_with_pretty_exceptions }.to raise_error(SystemExit)
     end
 
+    context "impact of concurrency value" do
+      before(:each) do
+        allow(Chef::Knife::WinrmSession).to receive(:new).and_return(session)
+        @winrms = Chef::Knife::Winrm.new(['-C', '3', '-m', 'localhost', '-x', 'testuser', '-P', 'testpassword', '--winrm-authentication-protocol', 'basic', 'echo helloworld'])
+      end
+      it "connect sessions if concurrent option is given" do
+        @winrms.config[:concurrency] = 1
+        allow(@winrms).to receive(:relay_winrm_command).and_return(true)
+        expect { @winrms.run_with_pretty_exceptions }.not_to raise_error
+      end
+      it "connect sessions if concurrent option is not given" do
+        allow(@winrm).to receive(:relay_winrm_command).and_return(true)
```

Comment:
We've set -C 3 in the before block, so wouldn't it still be given here?  We might need to `@winrms.config.delete[:concurrency]`.